packages/ckeditor5-link/src/linkcommand.ts

```diff
@@ -277,4 +291,46 @@ export default class LinkCommand extends Command {
 
 		return true;
 	}
+
+	/**
+	 * Updates selected link with a new value as its content and as its href attribute.
+	 *
+	 * @param model Model is need to insert content.
+	 * @param writer Writer is need to create text element in model.
+	 * @param range A range where should be inserted content.
+	 * @param href A link value which should be in the href attribute and in the content.
+	 */
+	private _updateLinkContent( model: Model, writer: Writer, range: Range, href: string ): Range {
+		const text = writer.createText( href, { linkHref: href } );
+
+		return model.insertContent( text, range );
+	}
+}
+
+/**
+	 * Returns text from the given range.
+	 *
+	 * @param range A range which text is returns from.
+	 */
+function getTextFromRange( range: Range | null ): string | null {
+	if ( !range || range && !Array.from( range.getItems() ).length ) {
+		return '';
+	}
+
+	const firstNode = Array.from( range.getItems() )[ 0 ];
+
+	if ( firstNode.is( '$text' ) || firstNode.is( '$textProxy' ) ) {
+		return firstNode.data;
+	}
+
+	return null;
+}
+
+function extractTextFromSelection( selection: DocumentSelection ): string | null {
+	const range = selection.getFirstRange( );
+	const firstPosition = selection.getFirstPosition();
+	const textFromPosition = firstPosition !== null && firstPosition.textNode && firstPosition.textNode.data || null;
```

Comment:
You can safely assume `firstPosition`[ always exists](https://github.com/ckeditor/ckeditor5/pull/13702/files#r1143111217) and use `const textFromPosition = firstPosition!.textNode && firstPosition!.textNode.data`.